src/xbot/common/math/PID.java

```diff
@@ -23,6 +28,37 @@ public void reset()
     {
         m_prevError = 0;
         m_totalError = 0;
+        errorIsSmall = false;
+        derivativeIsSmall = false;
+    }
+    
+    /**
+     * Set how close the error can be before it is considered
+     * "on-target."
+     *            
+     * This is in the same units as your current and goal values.
+     */
+    public void setErrorTolerance(double errorTolerance) {
```

Comment:
For tests, we can use the RobotAssertionManager to make sure nobody is setting negative values by throwing exceptions in their tests, but not sure what the best runtime option is. Might just be silently continue (akin to when you isOnTarget with no tolerances?)